src/composables/auth.ts

```diff
@@ -0,0 +1,117 @@
+import { useModals } from './modals';
+
+let composableInitialized = false;
+const isAvailable = ref(false);
+const isAuthenticated = ref(false);
+
+export function useAuth() {
+  const { isSecureLoginEnabled } = useUi();
+  const { openModal } = useModals();
+
+  async function checkIsAvailable() {
+    isAvailable.value = (await BiometricAuth.checkBiometry()).isAvailable;
+  }
+
+  /**
+   * Prompts the user to authenticate using biometric authentication.
+   * Returns a promise that resolves when the user is authenticated
+   * or if biometric authentication is not available.
+   */
+  async function authenticate(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      if (!isAvailable.value || !isSecureLoginEnabled.value) {
+        resolve();
+        return;
+      }
+      try {
+        resolve(BiometricAuth.authenticate({
+          reason: t('biometricAuth.reason'),
+          cancelTitle: t('common.cancel'),
+          allowDeviceCredential: true,
+          iosFallbackTitle: t('biometricAuth.fallbackTitle'),
+          androidTitle: t('biometricAuth.title'),
+          androidSubtitle: t('biometricAuth.subtitle'),
+          androidConfirmationRequired: false,
+        }).then(() => {
+          isAuthenticated.value = true;
+        }));
+      } catch (error) {
+        reject(error);
+      }
+    });
+  }
+
+  function deauthenticate() {
+    isAuthenticated.value = false;
+  }
+
+  async function openAuthModal() {
+    await checkIsAvailable(); // Check if biometric auth is still available
+    if (isAvailable.value && isSecureLoginEnabled.value && !isAuthenticated.value) {
+      await openModal(MODAL_SECURE_LOGIN);
+    }
+  }
+
+  async function openEnableAuthModal() {
+    if (IS_MOBILE_APP) {
+      openModal(MODAL_ENABLE_SECURE_LOGIN);
+    }
+  }
+
+  (async () => {
+    if (composableInitialized) {
+      return;
+    }
+    composableInitialized = true;
+    await checkIsAvailable();
+
+    // Create persistent effect scope to avoid watcher being disposed
+    effectScope(true).run(() => {
+      const {
+        isSecureLoginEnabled: scopedIsSecureLoginEnabled,
+        isAppActive,
+        lastTimeAppWasActive,
+        secureLoginTimeout,
+      } = useUi();
+
+      watch(
+        isAppActive,
+        async (isActive, wasActive) => {
+          // App resumed from background
+          if (scopedIsSecureLoginEnabled.value && isActive && !wasActive && IS_MOBILE_APP) {
+            if (isAuthenticated.value && lastTimeAppWasActive.value) {
+              const elapsedTime = Date.now() - lastTimeAppWasActive.value;
+              if (elapsedTime > secureLoginTimeout.value) {
+                deauthenticate();
+                await openAuthModal();
+              }
+            } else if (!isAuthenticated.value) {
+              await openAuthModal();
+            }
+          }
+        },
+      );
+    });
+  })();
+
+  return {
+    isAvailable: readonly(isAvailable),
```

Comment:
I think this should not be available from the outside as we are not always sure if we already set the value.